plugins/ksp/src/org/jetbrains/kotlin/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -288,21 +289,19 @@ class ResolverImpl(
                 typeReference.lookup()?.let {
                     return getKSTypeCached(it, type.element.typeArguments, type.annotations)
                 }
-                val containingDeclaration = KtStubbedPsiUtil.getContainingDeclaration(typeReference)
-                return if (containingDeclaration != null) {
+                KtStubbedPsiUtil.getContainingDeclaration(typeReference)?.let { containingDeclaration ->
                     resolveDeclaration(containingDeclaration)?.let {
                         // TODO: only resolve relevant branch.
                         ForceResolveUtil.forceResolveAllContents(it)
                     }
                     typeReference.lookup()?.let {
-                        getKSTypeCached(it, type.element.typeArguments, type.annotations)
-                    }
-                } else {
-                    val scope = resolveSession.fileScopeProvider.getFileResolutionScope(typeReference.containingKtFile)
-                    resolveSession.typeResolver.resolveType(scope, typeReference, bindingTrace, false).let {
-                        getKSTypeCached(it, type.element.typeArguments, type.annotations)
+                        return getKSTypeCached(it, type.element.typeArguments, type.annotations)
                     }
                 }
+                val scope = resolveSession.fileScopeProvider.getFileResolutionScope(typeReference.containingKtFile)
+                return resolveSession.typeResolver.resolveType(scope, typeReference, bindingTrace, false).let {
+                    getKSTypeCached(it, type.element.typeArguments, type.annotations)
+                }
```

Comment:
This block becomes a fall-through when the above resolution fails. It only works for references to symbols defined in file or package scope.

Reply:
Ah yes, that's the difference. But looks like this is working for the test case.

Reply:
Alright, please add a todo to revert this.